packages/tenant-process/src/services/tenantService.ts

```diff
@@ -729,13 +729,29 @@ export function tenantServiceBuilder(
           correlationId
         );
 
-        await repository.createEvents([
+        const createdEvents = await repository.createEvents([
           tenantCertifiedAttributeRevokedEvent,
           tenantKindUpdatedEvent,
         ]);
-      } else {
-        await repository.createEvent(tenantCertifiedAttributeRevokedEvent);
+
+        const newVersion = createdEvents.reduce(
+          (acc, event) => Math.max(acc, event.newVersion),
+          0
+        );
```

Comment:
More readable using map IMO

```suggestion
const newVersion = Math.max(0, ...createdEvents.map(event => event.newVersion));
```